07_VirtualFileSystem/03_TarFileSystem.md

```diff
@@ -0,0 +1,158 @@
+
+The _name of linked file_ field refers to symbolic links in the unix world, when a file is a link to another file, that field contains the value of the target file of the link.
+
+The USTar indictator (containing the string `ustar` followed by NULL), and the version field are used to identify the format being used, and the version field value is "00". 
+
+The `filename prefix` field, present only in the `ustar`, this format allows for longer file names, but it is splitted into two parts the `file name` field ( 100 bytes) and the `filename prefix` field (155 bytes)
+
+The other fields are either self-explanatory (like uid/gid) or can be left as 0 (TO BE CHECKED) the only one that needs more explanation is the `file size` field because it is expressed  as an octal number encoded in ASCII. This means we need to convert an ascii octal into a decimal integer. Just to remind, an `octal` number is a number represetend in base 8, we can use digits from 0 to 7 to represent it (is like binaries that use only 0 and 1 or hexadecimals that use from 0 to F). So for example:
+
+```
+octal 12 = hex A = bin 1010
+```
+
+In C an octal number is represented adding a `0` in front of the number, so for example 0123 is 83 in decimal.
+
+But that's not all, we also have that the number is represented as an `ascii` characters, so to get the decimal number we need to: 
+
+1. Convert each ansii digit into decimal, this should be pretty easy to do, since in the ascii table the digits are placed in ascending order starting from 0x30 ( `´0'` ), to get the digit we need just to subrstract the `ascii` code for the 0 to the char supplied
+2.  To obtain the decimal number from an octal we need to multiply each digit per `8^i` where i is the digit position (rightmost digit is 0) and sum their results. For example 37 in octal is: 
+
+```c
+037 = 3 * 8^1 + 7 * 8^0 = 31
+```
+
+Remember we ignore the first 0 because it tells C that it is an octal number (and also it doesn't add any value to the final result!), since we are writing an os implementing this function should be pretty easy, so this will be left as exercise, we will just assume that we have the following function available to convert octal ascii to decimal: 
+
+
+```c
+int octascii_to_dec(char *number, int size);
+```
+
+The size parameter tells us how many bytes is the digit long, and in the case of a tar object record the size is fixed: 12 bytes. Since we just need to implement a data structure for the header, this is left as exercise. Let's assume just that a new type is defined with the name `tar_record`.
+
+
+### Searching for a file
+
+Since the tar format doesn't have any file table or linked lists, or similar to search for files, we need everytime to start from the first record and scan one after each other, if the record is found we will return the pointer to it, otherwise we will eventually reach the end of the archive (file system in our case) meaning that the file searched is not present. 
```

Comment:
I would mention that in your (the readers) design they may want to create a faster structure to lookup file locations. Maybe a linked list with each entry containing the filename, size and a pointer to where it's data starts in the tar image. That would drastically speed up searching.

Reply:
of course we dont have to implement it, but it might to hint to the reader